packages/gafl-webapp-service/src/processors/__tests__/pricing-summary.spec.js

```diff
@@ -1,347 +1,562 @@
-import { dobHelper, JUNIOR_TODAY, ADULT_TODAY, SENIOR_TODAY } from '../../__mocks__/test-utils-business-rules.js'
-import { start, stop, initialize, injectWithCookies, mockSalesApi } from '../../__mocks__/test-utils-system.js'
-import { licenseTypes } from '../../pages/licence-details/licence-type/route.js'
+import { pricingDetail } from '../pricing-summary.js'
 
-import {
-  LICENCE_TO_START,
-  DATE_OF_BIRTH,
-  LICENCE_TYPE,
-  CONTROLLER,
-  DISABILITY_CONCESSION,
-  GET_PRICING_TYPES,
-  GET_PRICING_LENGTHS
-} from '../../uri.js'
-
-import { licenceToStart } from '../../pages/licence-details/licence-to-start/update-transaction.js'
-import { disabilityConcessionTypes } from '../../pages/concessions/disability/update-transaction.js'
-
-mockSalesApi()
-
-beforeAll(() => new Promise(resolve => start(resolve)))
-beforeAll(() => new Promise(resolve => initialize(resolve)))
-afterAll(d => stop(d))
-
-const juniorPricingByType = {
-  [licenseTypes.troutAndCoarse2Rod]: {
-    '12M': {
-      cost: 0,
-      concessions: true
-    },
-    msg: 'no-short'
-  },
-  [licenseTypes.troutAndCoarse3Rod]: {
-    '12M': {
-      cost: 0,
-      concessions: true
-    },
-    msg: 'no-short'
-  },
-  [licenseTypes.salmonAndSeaTrout]: {
-    '12M': {
-      cost: 0,
-      concessions: true
+jest.mock('../filter-permits.js', () => ({
```

Comment:
one comment - is it possible to create a createPermit function for example and pass in parameters that change instead repeating the creation?

Reply:
Not really - if you look at each item, there's enough variation that it doesn't really make sense to have such a method. If you did, it would end up not saving you anything.